pkg/resources/csi/vmwareclouddirector/csi.go

```diff
@@ -81,3 +84,29 @@ func GetVMwareCloudDirectorCSIConfig(cluster *kubermaticv1.Cluster, dc *kubermat
 
 	return cc.toString()
 }
+
+// SecretCreator returns the CSI secrets for vmwareclouddirector.
+func SecretCreator(data *resources.TemplateData) reconciling.NamedSecretCreatorGetter {
```

Comment:
could you split this file to two different files?
`SecretCreator` should be in `secret.go` and config in `config.go`

The same could apply to vsphere.